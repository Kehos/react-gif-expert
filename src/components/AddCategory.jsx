import { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ onNewValue }) => {

  const [inputValue, setInputValue] = useState('')

  // Rebem un 'event' i desestructurem per obtenir nomÃ©s 'target'
  const onInputChange = ({ target }) => {
    setInputValue(target.value)
  }

  const onSubmit = (event) => {
    event.preventDefault();
    if (inputValue.trim().length <= 1) return
    onNewValue(inputValue.trim())
    setInputValue('')
  }

  return (
    <form onSubmit={ onSubmit } aria-label='gif-form'>
      <input
        type="text"
        placeholder="Buscar gifs"
        value={ inputValue }
        onChange={ onInputChange }
      />
    </form>
  )
}

AddCategory.propTypes = {
  onNewValue: PropTypes.func.isRequired
}
