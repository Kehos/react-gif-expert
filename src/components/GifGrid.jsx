import { GifItem } from './GifItem';
import { useFetchGifs } from '../hooks/UseFetchGifs';
import PropTypes from 'prop-types';

export const GifGrid = ({ category }) => {

  // Custom hook useFetchGifs()
  // Podem ficar dins un custom hook tota la lògica de useState o useEffect
  const { images, isLoading } = useFetchGifs(category);

  return (
    <>
      <h3>{ category }</h3>
      {
        // En comptes de fer un condicional ternari fem això si no tenim
        // opció de "else"
        isLoading && <h2>Cargando...</h2>
      }

      <div className='card-grid'>
        {
          images.map((image) => (
            <GifItem
              key={ image.id }
              // Amb el property spread podem passar tots els camps d'un element
              // sense haver de passar-los per seaparat, com en aquest cas hauria
              // de ser "title={ image.title } url={ image.url }"
              { ...image }
            />
          ))
        }
      </div>
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}
