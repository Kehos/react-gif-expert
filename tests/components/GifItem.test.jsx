import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('GifItem component', () => {
  const mockTitle = 'Gif title';
  const mockUrl = 'http://url';

  test('Must match with snapshot', () => {
    const { container } = render(<GifItem title={ mockTitle } url={ mockUrl } />);

    expect(container).toMatchSnapshot();
  })

  test('Must show image with URL and ALT attributes', () => {
    render(<GifItem title={ mockTitle } url={ mockUrl } />);
    const { alt, src } = screen.getByRole('img');

    expect(alt).toBe(mockTitle);
    expect(src).toBe(`${mockUrl}/`);
  })

  test('Must show title into component', () => {
    render(<GifItem title={ mockTitle } url={ mockUrl } />);

    expect(screen.getByText(mockTitle)).toBeTruthy();
  })
})