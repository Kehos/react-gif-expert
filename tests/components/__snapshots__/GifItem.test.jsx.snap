// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem component Must match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Gif title"
      src="http://url"
    />
    <p>
      Gif title
    </p>
  </div>
</div>
`;
